src/sql/parts/objectExplorer/viewlet/serverTreeController.ts

```diff
@@ -78,7 +79,10 @@ export class ServerTreeController extends treedefaults.DefaultController {
 		if (element instanceof TreeNode) {
 			let context = new ObjectExplorerActionsContext();
 			context.nodeInfo = element.toNodeInfo();
+			// Note: getting DB name before, but intentionally not using treeUpdateUtils.getConnectionProfile as it replaces
+			// the connection ID with a new one. This breaks a number of internal tasks
 			context.connectionProfile = element.getConnectionProfile().toIConnectionProfile();
+			context.connectionProfile.databaseName = element.getDatabaseName();
```

Comment:
This may not work for top level database nodes. Is that something we need to support?

Reply:
In that case the database is already set correctly. This path is only relevant for TreeNode objects, not ConnectionProfile objects.

Reply:
Also - I've verified all the existing internal actions work as expected. The reason I implemented this way was that a previous change caused them all to break. Long term, I think that with this change we could try removing a lot of `TreeUpdateUtils.getConnectionProfile` calls in each of those actions, since they all actually need this logic to get the right context.